src/llm_models/open_ai.rs

```diff
@@ -0,0 +1,371 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use log::info;
+use reqwest::{header, Client};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+
+use crate::{
+    constants::{OPENAI_API_URL, OPENAI_BASE_INSTRUCTIONS, OPENAI_FUNCTION_INSTRUCTIONS},
+    domain::{OpenAPIChatResponse, OpenAPICompletionsResponse, RateLimit},
+    llm_models::LLMModel,
+};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub enum OpenAIModels {
+    Gpt3_5Turbo,
+    Gpt3_5Turbo0613,
+    Gpt3_5Turbo16k,
+    Gpt4,
+    Gpt4_32k,
+    TextDavinci003,
+    Gpt4Turbo,
+}
+
+#[async_trait(?Send)]
+impl LLMModel for OpenAIModels {
```

Comment:
This is another case when even though the entire module looks "new" it is actually mostly a copy/paste of the pre-existing `models` module just modified to implement the `LLMModels` trait.